gef.py

```diff
@@ -11133,27 +11149,32 @@ def __setup_qemu(self) -> bool:
         return True
 
     def __setup_remote(self) -> bool:
+        success = True
+
         # get the file
-        fpath = f"/proc/{self.pid}/exe"
-        if not self.sync(fpath, str(self.file)):
-            err(f"'{fpath}' could not be fetched on the remote system.")
-            return False
+        if not self.sync(str(self.file)):
+            warn(f"'{fpath}' could not be fetched on the remote system.")
+            success = False
 
-        # pseudo procfs
-        for _file in ("maps", "environ", "cmdline"):
-            fpath = f"/proc/{self.pid}/{_file}"
-            if not self.sync(fpath):
-                err(f"'{fpath}' could not be fetched on the remote system.")
-                return False
+        if not self.sync(f"/proc/{self.pid}/maps"):
+            warn(f"'{fpath}' could not be fetched on the remote system.")
+            success = False
```

Comment:
```suggestion
        try:
            self.sync(f"/proc/{self.pid}/maps")
        except gdb.error as e:
            warn(f"sync('{fpath}' failed, reason: {str(e)}")
            [...] setup the mock /proc/maps here[...]
```

Reply:
Waiting for what you think about this before changing anything :)
https://github.com/hugsy/gef/pull/1045#discussion_r1472057991